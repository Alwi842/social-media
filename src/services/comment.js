import axios from "axios";
export const getCommentById = async (id) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/comments/?postId=${id}`);
    return response.data;
  } catch (err) {
    throw new Error("failed to fetch data: ", err);
  }
};
