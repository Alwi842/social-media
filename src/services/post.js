import axios from "axios";
export const getPostById = async (id) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/posts/${id}`);
    return response.data;
  } catch (err) {
    throw new Error("failed to fetch data: ", err);
  }
};
