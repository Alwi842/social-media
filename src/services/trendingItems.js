export const getTrendingItems = async () => {
  //fake trending
  const trendingItems = [
    "React 18 Released",
    "JavaScript Tips",
    "Next.js vs Gatsby",
    "Web3 Development",
    "CSS Grid Layout",
    "AI in Web Development",
    "Frontend Frameworks",
    "Node.js Performance",
  ];
  return trendingItems;
};
