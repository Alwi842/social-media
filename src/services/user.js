import axios from "axios";

export const getUserById = async (userId) => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/users/${userId}`);

    // Return user data with the avatar
    return {
      [userId]: {
        ...response.data, // Add fetched user data
        avatar: `https://picsum.photos/id/${userId}/100`, // Add avatar
      },
    };
  } catch (error) {
    // Log error and return an empty object or handle the error as needed
    console.error("Error fetching user:", error);
    return {}; // Return an empty object to avoid crashing the app
  }
};
export const getUsers = async () => {
  try {
    const response = await axios.get(`https://jsonplaceholder.typicode.com/users/`);

    // Iterate through the response data and add avatar for each user
    const usersWithAvatars = response.data.map((user) => ({
      ...user, // Spread existing user data
      avatar: `https://picsum.photos/id/${user.id}/100`, // Add avatar based on user id
    }));

    // Return the users with avatar data, using their id as the key
    const usersById = usersWithAvatars.reduce((acc, user) => {
      acc[user.id] = user;
      return acc;
    }, {});

    return usersById;
  } catch (error) {
    // Log error and return an empty object or handle the error as needed
    console.error("Error fetching users:", error);
    return {}; // Return an empty object to avoid crashing the app
  }
};
