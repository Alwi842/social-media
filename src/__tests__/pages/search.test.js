import { render } from "@testing-library/react";
import SearchPage from "@/pages/search";
import "@testing-library/jest-dom";
import { getTrendingItems } from "@/services/trendingItems";
import { useRouter } from "next/router";

// Mock Next.js router
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

// Mock `getTrendingItems` service
jest.mock("@/services/trendingItems", () => ({
  getTrendingItems: jest.fn(),
}));

describe("Search Page", () => {
  beforeEach(() => {
    // Mock router to prevent "NextRouter was not mounted" error
    useRouter.mockReturnValue({
      pathname: "/search",
      query: {},
      push: jest.fn(),
    });

    // Mock API response (array of strings)
    getTrendingItems.mockResolvedValue([
      "React 18 Released",
      "JavaScript Tips",
      "Next.js vs Gatsby",
      "Web3 Development",
      "CSS Grid Layout",
      "AI in Web Development",
      "Frontend Frameworks",
      "Node.js Performance",
    ]);
  });

  it("renders correctly and matches snapshot", () => {
    const { container } = render(
      <SearchPage
        data={[
          "React 18 Released",
          "JavaScript Tips",
          "Next.js vs Gatsby",
          "Web3 Development",
          "CSS Grid Layout",
          "AI in Web Development",
          "Frontend Frameworks",
          "Node.js Performance",
        ]}
      />
    );

    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});
