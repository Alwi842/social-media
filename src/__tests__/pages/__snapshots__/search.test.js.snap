// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="bg-black p-4 space-y-6 h-full min-h-screen"
  >
    <div
      class="w-full h-full text-white p-4 space-y-6"
    >
      <button
        class="flex gap-2"
      >
        <svg
          fill="#e8eaed"
          height="24px"
          viewBox="0 -960 960 960"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
          />
        </svg>
        Back
      </button>
      <div
        class="flex items-center bg-gray-800 p-2 rounded-md"
      >
        <input
          class="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
      <div
        class="text-xl font-semibold"
      >
        Trending
      </div>
      <div
        class="space-y-2"
      >
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          React 18 Released
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          JavaScript Tips
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Next.js vs Gatsby
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Web3 Development
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          CSS Grid Layout
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          AI in Web Development
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Frontend Frameworks
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Node.js Performance
        </div>
      </div>
    </div>
  </div>
</div>
`;
